### PR TITLE
feat(UI): Apply user roles on dashboards actions

### PR DESCRIPTION
**Pull request by Thebarda**
## Description

This applies the current user roles on dashboard actions. It displays buttons and informations according to the user roles.

Jira : 
> 🏷️ **MON-18517** about `GET /configuration/dashboards` + `GET /configuration/dashboard/:id`
> 🏷️ **MON-18519** about `GET /configuration/users/current/parameters`

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by abuathier — DISMISSED** [review]

**Review by g10 — APPROVED** [review]

File: centreon/packages/ui-context/src/types.ts [hunk 1]
```diff
@@ -5,8 +5,22 @@ export enum ListingVariant {
...
+export interface DashboardRolesAndPermissions {
+  createDashboards: boolean;
+  globalUserRole: DashboardGlobalRole;
+  manageAllDashboards: boolean;
+  viewDashboards: boolean;
+}
+
 export interface User {
   alias: string;
+  dashboard: DashboardRolesAndPermissions;
```
Comment (g10) [inline centreon/packages/ui-context/src/types.ts:23]:
```suggestion
  dashboard?: DashboardRolesAndPermissions;
```

File: centreon/www/front_src/src/api/decoders.ts [hunk 2]
```diff
@@ -1,13 +1,38 @@
...
+    globalUserRole: 'global_user_role',
+    manageAllDashboards: 'administrate_role',
+    viewDashboards: 'view_role'
+  }
+);
+
 export const userDecoder = JsonDecoder.object<User>(
   {
     alias: JsonDecoder.string,
+    dashboard: dashboardDecoder,
```
Comment (g10) [inline centreon/www/front_src/src/api/decoders.ts:35]:
```suggestion
    dashboard: JsonDecoder.optional(dashboardDecoder),
```

**Review by g10 — CHANGES_REQUESTED** [review]
fyi. `/configuration/users/current/parameters` does not include a `dashboard` prop (...in this branch), therefore the failing tests

**Comment by Thebarda** [comment]
@abuathier Where are the changes for `/configuration/users/current/parameters`?

File: centreon/www/front_src/src/api/decoders.ts (hunk 2 above)
Comment (Thebarda) [reply centreon/www/front_src/src/api/decoders.ts:35]: I just don't want to make something optional if it's temporary

**Review by g10 — APPROVED** [review]

**Review by abuathier — DISMISSED** [review]

**Review by g10 — APPROVED** [review]

**Review by abuathier — APPROVED** [review]

**Pull request merged**